src/cmd/bootstrap.ts

```diff
@@ -158,80 +160,101 @@ const bootstrapValues = async (): Promise<void> => {
       '`otomi.adminPassword` has been generated and is stored in the values repository in `env/secrets.settings.yaml`',
     )
   }
+}
 
+const postSops = async (): Promise<void> => {
   if (existsSync(`${env.ENV_DIR}/.sops.yaml`)) {
     // encryption related stuff
     const file = '.gitattributes'
     await copyFile(`${rootDir}/.values/${file}`, `${env.ENV_DIR}/${file}`)
     // just call encrypt and let it sort out what has changed and needs encrypting
     await encrypt()
   }
+}
 
-  if (!hasOtomi) {
-    debug.log('You can now use the otomi CLI')
+const customCA = async (originalValues: Record<string, any>): Promise<void> => {
+  const d = terminal('customCA')
+  const cm = get(originalValues, 'charts.cert-manager', {})
+
+  if (cm?.customRootCA && cm?.customRootCAKey) {
+    d.info('Skipping custom RootCA generation')
+    return
   }
-  debug.log(`Done bootstrapping values`)
+  d.info('Need to generate custom RootCA')
+
+  // Code example from: https://www.npmjs.com/package/node-forge#x509
+  const keys = pki.rsa.generateKeyPair(2048)
+  const cert = pki.createCertificate()
+  cert.setExtensions([
+    {
+      name: 'basicConstraints',
+      cA: true,
+    },
+    {
+      name: 'keyUsage',
+      keyCertSign: true,
+      digitalSignature: true,
+      nonRepudiation: true,
+      keyEncipherment: true,
+      dataEncipherment: true,
+    },
+  ])
+  cert.publicKey = keys.publicKey
+  cert.serialNumber = '01'
+  cert.validity.notBefore = new Date()
+  cert.validity.notAfter = new Date()
+  cert.validity.notAfter.setFullYear(cert.validity.notBefore.getFullYear() + 10)
+  const attrs = [
+    { name: 'commonName', value: 'redkubes.com' },
```

Comment:
```suggestion

```

CN is (considered) deprecated.

Reply:
["The domain of the URL must match the subject of the certificate. In former times this could be either by setting the domain as CN of the certificate or by having the domain set as a subject alternative name. Support for CN was deprecated for a long time (at least 17 years, see RFC 2818) and Chrome browser will not even look at the CN anymore so today you need to have the domain of the URL as a subject alternative name. Note that there can be multiple subject alternative names and thus the certificate can be used for multiple domains." ](https://security.stackexchange.com/questions/175786/is-it-required-to-have-the-same-domain-name-and-common-name-for-ssl-certificate?rq=1)

Reply:
so it can be removed indeed

Reply:
Sort of, if you read closely it is replaced by SAN. Also, it varies depending on what checks the certificate and how RFC-compliant it is, CN will sometimes be read if there is no SAN.

Reply:
ok